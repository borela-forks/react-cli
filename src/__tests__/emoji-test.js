// @flow strict

import * as React from "react";
import ReactCLI, { Section } from "../index";

test("columns containing emojis should render the correct width", done => {
  ReactCLI(
    <Section border={{ horizontal: "*", vertical: "*" }}>
      Row 1 <br /> Row 2 🎉🍾🎉
    </Section>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );

  ReactCLI(
    <Section border={{ horizontal: "*", vertical: "*" }}>
      Emoji test line one, its a lont line that takes space 🤘
      <br />
      🛠 Multiple emojis in the same line! 🍾 🛫
      <br />
      🛠 Some more emojis with elipsis... 🕸
      <br />
      An emoji in the middle 🛠 thats so {'"--crazy"'}
      <br />
      one more for good measure... 🕸
      <br />
    </Section>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );

  ReactCLI(
    <Section border={{ horizontal: "*", vertical: "*" }} align="center">
      ✔︎ Step 1<br />◯ Step 2<br />◯ Step 3
    </Section>,
    undefined,
    50,
    outputString => {
      expect(outputString).toMatchSnapshot();
      done();
    },
    "+"
  );
});
