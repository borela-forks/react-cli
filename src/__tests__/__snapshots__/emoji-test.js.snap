// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columns containing emojis should render the correct width 1`] = `
"**************************************************
*Row 1+++++++++++++++++++++++++++++++++++++++++++*
*Row 2 🎉🍾🎉++++++++++++++++++++++++++++++++++++*
**************************************************
"
`;

exports[`columns containing emojis should render the correct width 2`] = `
"**************************************************
*Emoji test line one, its a lont line that takes+*
*space 🤘++++++++++++++++++++++++++++++++++++++++*
*🛠 Multiple emojis in the same line! 🍾 🛫++++++*
*🛠 Some more emojis with elipsis... 🕸++++++++++*
*An emoji in the middle 🛠 thats so \\"--crazy\\"++++*
*one more for good measure... 🕸+++++++++++++++++*
**************************************************
"
`;

exports[`columns containing emojis should render the correct width 3`] = `
"**************************************************
*+++++++++++++++++++✔︎ Step 1++++++++++++++++++++*
*++++++++++++++++++++◯ Step 2++++++++++++++++++++*
*++++++++++++++++++++◯ Step 3++++++++++++++++++++*
**************************************************
"
`;
